lib/app.js

```diff
@@ -51,17 +52,20 @@ require('fh-wfm-user/lib/router/cloud')(mediator, app, authServiceGuid);
 require('fh-wfm-workflow/lib/server')(mediator, app, mbaasApi);
 require('fh-wfm-workorder/lib/server')(mediator, app, mbaasApi);
 
-// app modules
-require('./app/message')(mediator);
-require('./app/workorder')(mediator);
-require('./app/result')(mediator);
-require('./app/workflow')(mediator);
-require('./app/group')(mediator);
-require('./app/file')(mediator);
-
 // fhlint-end
 
-// Important that this is last!
-app.use(mbaasExpress.errorHandler());
 
-module.exports = exports = app;
+module.exports = function() {
+  //make sure that all app modules finish setting up
+  return Promise.all([
+    require('./app/message')(mediator),
+    require('./app/workorder')(mediator),
+    require('./app/result')(mediator),
+    require('./app/workflow')(mediator),
+    require('./app/group')(mediator),
+    require('./app/file')(mediator)]).then(function() {
+      // Important that this is last!
```

Comment:
This comment is actually about the error handler middleware below this line, see indentation.

Middleware chain setup is sensitive to the order they're give to the express app.